Feature/catalog page card section

AlinaPalyvoda opened the pull request:
- JSX,
- Styles

AlinaPalyvoda committed:
feat: added CatalogTourCard

AlinaPalyvoda committed:
fix: changed the button in the tour card

AlinaPalyvoda committed:
fix: some syntax fixes

HIMDanny commented on client/src/components/CatalogTourCard/CatalogTourCard.module.scss:
```diff
@@ -0,0 +1,13 @@
+$primaryMainColor: #0499dd;
```
You can use mui styled component to take advantage of theme colors

Please, read this article in documentation:
https://mui.com/material-ui/customization/how-to-customize/

AlinaPalyvoda replied:
done

HIMDanny commented on client/src/components/CatalogTourCard/CatalogTourCard.module.scss:
```diff
@@ -0,0 +1,13 @@
+
+.cardBtn {
+  &:hover {
+    color: $secondaryMainColor !important;
```
Using styled mui components, mentioned above, you can also get rid of the `!important `rule

AlinaPalyvoda replied:
done

HIMDanny commented on client/src/pages/CataloguePage/CataloguePage.js:
```diff
@@ -1,5 +1,18 @@
+const CataloguePage = () => (
+  <Box sx={{ backgroundColor: '#EDEDED' }}>
+    <Container>
+      <Typography sx={{ marginBottom: '40px' }} variant="h2">TOURS</Typography>
```
What about using `text-transform: uppercase` in stylings instead of writing upper case in html?

AlinaPalyvoda replied:
done

HIMDanny commented on client/src/pages/CataloguePage/CataloguePage.js:
```diff
@@ -1,5 +1,18 @@
 import React from 'react';
+import { Stack, Container, Box, Typography } from '@mui/material';
+import CatalogTourCard from '../../components/CatalogTourCard/CatalogTourCard';
```
I suggest adding `CatalogTourCard` to index.js in components folder and make import like this:
```suggestion
import { CatalogTourCard } from '../../components';
```

AlinaPalyvoda replied:
done

HIMDanny commented on client/src/components/CatalogTourCard/CatalogTourCard.js:
```diff
@@ -0,0 +1,91 @@
+        <Typography
+          sx={{ color: 'black', marginBottom: '20px', marginTop: '20px' }}
+          variant='h3'
+          component='div'
```
I think it is better to leave this component to be `<h3>`, or change to `<p>`. To my mind, `<div>` is not semantically correct tag to use with text

Ahmed-Fahmy-Dev replied:
agree to the above and would like to add that you could utilize the colors in our mui default theme instead of assigning them with manual value.   as far as i remember, we dont have black text in our design but a dark 282828 color which is in our theme in the palette object, please look for it.

AlinaPalyvoda replied:
done

HIMDanny commented on client/src/components/CatalogTourCard/CatalogTourCard.js:
```diff
@@ -0,0 +1,91 @@
+        <Stack direction='row' spacing={3} alignItems='start' sx={{ marginBottom: '20px' }}>
+          <Stack direction='row' spacing={1} alignItems='center'>
+            <CoinsIcon />
+            <Typography variant='h3' color='primary'>
```
Don't think, that price should be `<h3>`. What about changing it to `<p>` or `<span>` component?

```suggestion
            <Typography variant='h3' color='primary' component='span'>
```

AlinaPalyvoda replied:
done

HIMDanny commented on client/src/components/CatalogTourCard/CatalogTourCard.js:
```diff
@@ -0,0 +1,91 @@
+
+          <Stack direction='row' spacing={1} alignItems='center'>
+            <AccessTimeIcon color='primary' sx={{ height: '21px' }} />
+            <Typography variant='h3' color='primary'>
```
Consider my previous comment about price
```suggestion
            <Typography variant='h3' color='primary' component='span'>
```

AlinaPalyvoda replied:
done

Ahmed-Fahmy-Dev commented on client/src/pages/CataloguePage/CataloguePage.js:
```diff
@@ -1,5 +1,18 @@
+    <Container>
+      <Typography sx={{ marginBottom: '40px' }} variant="h2">TOURS</Typography>
+      <Stack spacing={2}>
+        <CatalogTourCard />
+        <CatalogTourCard />
+        <CatalogTourCard />
```
this is unpractical in case of receiving hundreds of cards from the backend, please write a more production approach code utilizing the array map method.
you could create a dummy array consisting of card objects with a card header, card text, and card img url and use the map method to display them inside your stack.

AlinaPalyvoda replied:
This will be done later.
This is now presented for illustrative purposes.

Ahmed-Fahmy-Dev commented on client/src/components/CatalogTourCard/CatalogTourCard.js:
```diff
@@ -0,0 +1,91 @@
+  return (
+    <Stack
+      direction={{ xs: 'column', tablet: 'row' }}
+      sx={{ backgroundColor: 'white', borderRadius: '10px', boxShadow: '0px 0px 40px rgba(0, 0, 0, 0.05)' }}
```
white color could be used from our theme.

AlinaPalyvoda replied:
Done

Ahmed-Fahmy-Dev commented on client/src/components/CatalogTourCard/CatalogTourCard.js:
```diff
@@ -0,0 +1,91 @@
+        <Stack direction='row' spacing={3} alignItems='start' sx={{ marginBottom: '20px' }}>
+          <Stack direction='row' spacing={1} alignItems='center'>
+            <CoinsIcon />
+            <Typography variant='h3' color='primary'>
```
here you go,  you are using color : primary which is mui theme syntax.  but on the other hand you could delete the color property here since h3 is already styled by the primary color by default in our theme

AlinaPalyvoda replied:
done

Ahmed-Fahmy-Dev commented on client/src/components/CatalogTourCard/CatalogTourCard.js:
```diff
@@ -0,0 +1,91 @@
+          </Stack>
+        </Stack>
+
+        <Typography className={styles.tourDescription} sx={{ color: 'black' }} variant='body1' color='text.secondary'>
```
as mentioned above we dont have pure black in our design.  its 282828  and it is assigned to the following object key :      text.primary

AlinaPalyvoda replied:
done

Ahmed-Fahmy-Dev commented on client/src/components/CatalogTourCard/CatalogTourCard.js:
```diff
@@ -0,0 +1,91 @@
+
+        <CardActions>
+          <Button
+            className={styles.cardBtn}
```
consider using a styled component instead of this approach.  please feel free to contact me for more clarification.

AlinaPalyvoda replied:
done

Ahmed-Fahmy-Dev reviewed (commented):
please move the jpg image to the public/assets/images/cataloguePage/

AlinaPalyvoda committed:
Merge branch 'develop' into feature/catalog-page-card-section

AlinaPalyvoda committed:
feat: catalog page cards section:

- component added to index.js
- styling with material ui
- scss removed
- components are corrected according to the rules of semantics

AlinaPalyvoda committed:
fix: applied bg color from theme

HIMDanny commented on client/src/components/CatalogTourCard/CatalogTourCard.js:
```diff
@@ -0,0 +1,88 @@
+  marginBottom: 20,
+}));
+
+const TourDescription = styled(Typography)(({ theme }) => ({
```
You can add `variant="body1"` here, as you already using styled component
```suggestion
const TourDescription = styled((props) => <Typography variant="body1" {...props} />)(({ theme }) => ({
```

AlinaPalyvoda replied:
Done

HIMDanny commented on client/src/components/CatalogTourCard/CatalogTourCard.js:
```diff
@@ -0,0 +1,88 @@
+  },
+}));
+
+const CardTitle = styled(Typography)(({ theme }) => ({
```
```suggestion
const CardTitle = styled((props) => <Typography variant="h3" {...props} />)(({ theme }) => ({
```

AlinaPalyvoda replied:
Done

Ahmed-Fahmy-Dev committed:
Merge branch 'develop' into feature/catalog-page-card-section

Ahmed-Fahmy-Dev commented on client/src/components/CatalogTourCard/CatalogTourCard.js:
```diff
@@ -0,0 +1,88 @@
+        </Stack>
+      </Stack>
+
+      <TourDescription variant="body1">
```
```suggestion
      <TourDescription>
```

you can delete the variant body1 as it is by default included

AlinaPalyvoda replied:
Done

AlinaPalyvoda committed:
Merge branch 'develop' into feature/catalog-page-card-section

AlinaPalyvoda committed:
Merge branch 'feature/catalog-page-card-section' of https://github.com/AltysS/travelAroundUkraine into feature/catalog-page-card-section

AlinaPalyvoda committed:
Merge branch 'develop' into feature/catalog-page-card-section

AlinaPalyvoda committed:
fixes and optimization
